src/hooks/spindl/fetchSpindl.ts

```diff
@@ -13,51 +14,41 @@ export const fetchSpindl = async ({
   chainId,
   tokenAddress,
   address,
-}: Omit<SpindlFetchParams, 'enabled'>): Promise<
-  SpindlFetchResult | undefined
-> => {
+}: SpindlFetchParams): Promise<SpindlFetchResult | undefined> => {
   const spindlConfig = getSpindlConfig();
+  const locale = getLocale().split('-');
 
   if (!spindlConfig?.apiUrl || !spindlConfig.headers) {
     return;
   }
 
-  const url = new URL(`${spindlConfig.apiUrl}${FETCH_SPINDL_PATH}`);
-  url.searchParams.set('placement_id', 'notify_message');
-  url.searchParams.set('limit', '2');
-  const locale = getLocale().split('-');
-  if (address) {
-    // wallet address, in the form of 0x... (required)
-    url.searchParams.set('address', address);
-  }
-  if (country || locale.length) {
-    // country code (IN, US, etc.)
-    url.searchParams.set('country', country || locale[1]);
-  }
-  if (chainId) {
-    // (Optional): The chain id (numeric, from https://chainlist.org)
-    url.searchParams.set('chain_id', chainId.toString());
-  }
-  if (tokenAddress) {
-    // (Optional): The contract address of the token (on the chain specified above)
-    url.searchParams.set('token_address', tokenAddress);
-  }
-
-  const response = await fetch(url, {
-    method: 'GET',
-    headers: spindlConfig.headers,
-  });
+  const queryParams: Record<string, string> = {
+    placement_id: 'notify_message',
+    limit: '2',
+    address: address || '',
```

Comment:
should be ommited if not exists and not set to '' (need to verify on their side but might create some side effect)